test/parallel_testsuite.py

```diff
@@ -85,6 +87,9 @@ def run(self, result):
     # issues.
     # multiprocessing.set_start_method('spawn')
 
+    # Remove any old stale list of flaky tests before starting the run
+    utils.delete_file(common.flaky_tests_log_filename)
```

Comment:
Shouldn't this apply to regular, non-parallel test runner too?  (You can always use `-j1` to trigger the use of the non-parallel runner with any suite).